src/qubic.cpp

```diff
@@ -610,34 +615,66 @@ static void processBroadcastComputors(Peer* peer, RequestResponseHeader* header)
                 enqueueResponse(NULL, header);
             }
 
-            // Copy computor list
-            bs->CopyMem(&broadcastedComputors.computors, &request->computors, sizeof(Computors));
-
-            // Update ownComputorIndices and minerPublicKeys
-            if (request->computors.epoch == system.epoch)
+            if (useSelfGeneratedComputors)
             {
-                numberOfOwnComputorIndices = 0;
-                ACQUIRE(minerScoreArrayLock);
+                // Compare received computor list (signed by ARB) and self generated computor list
+                bool listMatches = true;
+
                 for (unsigned int i = 0; i < NUMBER_OF_COMPUTORS; i++)
                 {
-                    minerPublicKeys[i] = request->computors.publicKeys[i];
+                    if(selfGeneratedComputors[i] != request->computors.publicKeys[i]){
+                        listMatches = false;
+                        break;
+                    }
+                }
+                if (!listMatches)
+                {
+                    // Stop tickProcessor due to missmatch
+                    criticalSituation = 2;
+#ifndef NDEBUG
+                    addDebugMessage(L"Error: Computor list received from ARB does not match self-generated list. Raise criticalSituation #2.");
+#endif
+                }
+                else
+                {
+                    // Accept list of arbitrator and mark as signed
+                    bs->CopyMem(&broadcastedComputors.computors.signature, request->computors.signature, SIGNATURE_SIZE);

```

Comment:
Please use copyMem() in all new code. We want to reduce platform dependency as much as possible to simplify testing and reuse of code outside UEFI.